### PR TITLE
Jeffotter/tech blog/wordembeddings

### PR DESCRIPTION
**Pull request by Jeffotter**
Added in some new data ran it through chat GPT and added in some formatting

### CODE REVIEW

File: docs/blog/posts/2020-03-06-wordEmbeddings.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+---
+date:
+  created: 2020-03-04
+categories:
+    - Machine Learning
+    - GPU
+authors:
+    - nrapstin
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2020-03-06-wordEmbeddings.md:8]: i did not write this

File: docs/blog/.authors.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 authors:
   jeffotter:
     name: Jeff Ott        # Author name
-    description: Darc Team Research Analytics Scientist # Author description
+    description: 🍀 Darc Team Research Analytics Scientist # Author description
```
Comment (natalya-patrikeeva) [inline docs/blog/.authors.yml:4]: Format it similar to everyone else

File: docs/blog/posts/2020-03-06-wordEmbeddings.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+| Man   | 0.12351   | 0.067231  | 0.85921   | 0.821231  |
+| Woman | 0.23131   | 0.123151  | 0.011249  | 0.23151   |
+
+However, in real world examples, it is rarely this simple.  In general, we do not know exactly what each of the N features mean.  It is very common to test the quality of embeddings using analogies.  For example, even though we may not be able to identify a feature for Royalty, we expect that subtracting the vectors "King - Man" and "Queen - Woman" should give us roughly the same vector representing Royalty.  Similarily for Masculinity, we can test the similarity of the vectors "Man - Woman" and "King - Queen".  
+
+## Common Embedding Models
+
+
+### Word2Vec
+Developed by Google, it uses skip-gram and continuous bag-of-words (CBOW) architectures to create word embeddings. [You can read more on Word2Vec's wikipedia page.](https://en.wikipedia.org/wiki/Word2vec) [The Gensim package in python has an implementation of these algorithms](https://radimrehurek.com/gensim/models/word2vec.html)
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2020-03-06-wordEmbeddings.md:45]: make sure all links open up in new tabs. You are also missing a lot of end of sentence periods.

File: docs/blog/posts/2020-03-06-wordEmbeddings.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+## An Example
+
+| Word  | "Royalty" Feature | "Masculinity" Feature |
+| ----- | ----------------- | --------------------- |
+| Queen | 0.958123          | 0.03591294            |
+| King  | 0.94981289        | 0.92959219            |
+| Man   | 0.051231          | 0.9592109321          |
+| Woman | 0.0912987         | 0.04912983189         |
+
+Let's say we were to create embeddings for the four words above, with two features representing Royalty and Masculinity.  The resulting embedding for the word "Queen" is <0.958123, 0.03591294> - a value close to 1 for Royalty, and a value close to 0 for Masculinity.  In this example, we could identify the concept of Masculinity by the vector <0,1> and Royalty by <1,0>.
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2020-03-06-wordEmbeddings.md:30]: I don't know what <0,1> notation means

File: docs/blog/posts/2020-03-06-wordEmbeddings.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+| Word  | Feature 1 | Feature 2 | Feature 3 | Feature 4 |
+| ----- | --------- | --------- | --------- | --------- |
+| Queen | 0.854712  | 0.32145   | 0.123512  | 0.023051  |
+| King  | 0.9521    | 0.23151   | 0.721385  | 0.950213  |
+| Man   | 0.12351   | 0.067231  | 0.85921   | 0.821231  |
+| Woman | 0.23131   | 0.123151  | 0.011249  | 0.23151   |
+
+However, in real world examples, it is rarely this simple.  In general, we do not know exactly what each of the N features mean.  It is very common to test the quality of embeddings using analogies.  For example, even though we may not be able to identify a feature for Royalty, we expect that subtracting the vectors "King - Man" and "Queen - Woman" should give us roughly the same vector representing Royalty.  Similarily for Masculinity, we can test the similarity of the vectors "Man - Woman" and "King - Queen".  
+
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2020-03-06-wordEmbeddings.md:40]: it's not super clear. First, we convert string to numbers then you can subtract two vectors but they have to be equal lenght, etc. Saying "Man - Woman" might be confusing

File: docs/blog/posts/2020-03-06-wordEmbeddings.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+### Word2Vec
+Developed by Google, it uses skip-gram and continuous bag-of-words (CBOW) architectures to create word embeddings. [You can read more on Word2Vec's wikipedia page.](https://en.wikipedia.org/wiki/Word2vec) [The Gensim package in python has an implementation of these algorithms](https://radimrehurek.com/gensim/models/word2vec.html)
+
+### GloVe (Global Vectors for Word Representation)
+Developed by Stanford, it creates embeddings by aggregating global word-word co-occurrence statistics from a corpus. You can find GloVe and more information [**here**](https://nlp.stanford.edu/projects/glove/)
+
+### FastText
+Developed by Facebook, it extends Word2Vec by representing words as bags of character n-grams, which helps in handling out-of-vocabulary words.
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2020-03-06-wordEmbeddings.md:51]: what is out-of-vocabulary? you don't define this

File: docs/blog/posts/2020-03-06-wordEmbeddings.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+### GPT (Generative Pre-trained Transformer)
+Developed by OpenAI, it generates embeddings using transformer-based architectures, focusing on language generation tasks.
+
+### Universal Sentence Encoder
+Developed by Google, it creates embeddings for sentences and paragraphs, useful for tasks like semantic similarity and clustering.
+
+### API's
+
+Many companies offer API's to generate word embeddings.  [Google's Cloud Natural Language API](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-text-embeddings), [OpenAI](https://platform.openai.com/docs/guides/embeddings), and [Microsoft's Azure Cognitive Services](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/embeddings?tabs=console) all offer text analysis services that can generate word embeddings. If you are looking to use more open source embeddings there are model repositys like [Hugging Face](https://huggingface.co/models) that offer a wide range of models.
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2020-03-06-wordEmbeddings.md:67]: repositys typo

File: docs/blog/posts/2020-03-06-wordEmbeddings.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+## Common Embedding Uses
+
+### Regression and Classification
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2020-03-06-wordEmbeddings.md:71]: formatting does not look right on the web

File: docs/blog/posts/2020-03-06-wordEmbeddings.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
+---
+date:
+  created: 2020-03-04
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/2020-03-06-wordEmbeddings.md:3]: is this right?

**Review by natalya-patrikeeva — CHANGES_REQUESTED** [review]

**Review by natalya-patrikeeva — CHANGES_REQUESTED** [review]
looking good...one small rephrasing at the end of the blog

File: docs/blog/posts/wordEmbeddings.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+
+### Recommendation Systems
+Embeddings power modern recommenders through:
+
+- Content-based filtering
+- Item similarity calculation
+- User preference modeling
+
+## Implementation Tips
+- Consider using pre-trained embeddings like Word2Vec, GloVe, or BERT
```
Comment (natalya-patrikeeva) [inline docs/blog/posts/wordEmbeddings.md:116]: this should be "pre-trained embedding models"

Comment (natalya-patrikeeva) [reply docs/blog/posts/wordEmbeddings.md:116]: Let's add punctuation to the end of the three bullet points as well.

**Review by natalya-patrikeeva — CHANGES_REQUESTED** [review]
one last change

File: docs/blog/posts/wordEmbeddings.md (hunk 10 above)
Comment (Jeffotter) [reply docs/blog/posts/wordEmbeddings.md]: Done and Done!

**Review by natalya-patrikeeva — APPROVED** [review]
yay

**Pull request merged**